int32 cmd_x1
int32 cmd_x2
int32 cmd_y1
int32 cmd_y2
int32 cmd_z1
int32 cmd_z2